### PR TITLE
VBLOCKS-1912: Modify incoming call notification bring app to foreground upon click

### PR DESCRIPTION
**Pull request by afalls-twilio**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

Modified so that clicking on the incoming call notification brings up the application. Also, when the app is foregrounded with that notification, the notification minimizes into the notification tray.

## Breakdown

- Updated notification to be clickable
- registered view's on click handler to send event to hidden proxy activity.
- Added a dummy JS callback method so the JS app can know when this occurs.

## Validation

- Called app with activity out of focus and clicked on notification to bring application into focus.
- Called app with activity out of focus and clicked on notification buttons to ensure call was accepted or rejected.

## Additional Notes

Lets be better than whatsapp!

### CODE REVIEW

**Comment by kbagchiGWC** [comment]
@afalls-twilio What shows when notification body is tapped? Can you add few screenshots?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+1.0.0-beta.3 (In Development)
+============================
+
+Twilio Voice React Native SDK has now reached milestone `beta.3`. Included in this version are the following.
+
+## Features
+
+- Incomming call notifications can now be tapped to bring the application into the foreground.
```
Comment (mhuynh5757) [inline CHANGELOG.md:8]:
```suggestion
- Incoming call notifications can now be tapped to bring the application into the foreground.
```

**Review by mhuynh5757 — CHANGES_REQUESTED** [review]

File: android/src/main/res/layout/custom_notification_incoming.xml [hunk 2]
```diff
@@ -1,65 +1,69 @@
...
 
-</LinearLayout>
+      <Button
+        android:id="@+id/button_decline"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginRight="@dimen/activity_horizontal_margin"
+        android:layout_marginLeft="@dimen/activity_horizontal_margin"
+        android:layout_weight="0.5"
+        android:radius="10dp"
```
Comment (mhuynh5757) [inline android/src/main/res/layout/custom_notification_incoming.xml:47]: Noticed that this button has a radius but the answer button does not. Is that intentional?

Comment (afalls-twilio) [reply android/src/main/res/layout/custom_notification_incoming.xml:47]: No idea, I didn't modify that code, just moved it into a FrameLayout

**Review by mhuynh5757 — APPROVED** [review]

File: src/Voice.tsx [hunk 3]
```diff
@@ -375,6 +375,9 @@ export class Voice extends EventEmitter {
       [Constants.VoiceEventCallInviteRejected]: this._handleCallInviteRejected,
       [Constants.VoiceEventCallInviteCancelled]:
         this._handleCancelledCallInvite,
+      [Constants.VoiceEventCallInviteNotificationTapped]: () => {
+        /* TODO(mhuynh): handle this event */
```
Comment (kbagchiGWC) [inline src/Voice.tsx:379]: Do we need a ticket for this @mhuynh5757 @afalls-twilio?

Comment (mhuynh5757) [reply src/Voice.tsx:379]: Yeah probably a good idea. I've filed VBLOCKS-2017.

**Review by kbagchiGWC — APPROVED** [review]

**Pull request merged**